src/providers/dataProvider/tests/item.test.ts

```diff
@@ -0,0 +1,276 @@
+import {
+  type DataProvider,
+  type AuthProvider,
+  withLifecycleCallbacks
+} from 'react-admin'
+import localForageDataProvider from 'ra-data-local-forage'
+import { encryptedUsers } from '../../../utils/init-data'
+import authProvider from '../../authProvider'
+import { R_AUDIT, R_BATCHES, R_ITEMS, R_USERS } from '../../../constants'
+import { lifecycleCallbacks } from '..'
+import { trackEvent } from '../../../utils/audit'
+import {
+  generateDummyBatchForTesting,
+  generateItemForTesting
+} from './dummy-data'
+import { DateTime } from 'luxon'
+import { AuditType } from '../../../utils/activity-types'
+
+const TEST_STORAGE_KEY = 'rco-test'
+const year: number = 2025
+describe('CRUD operations on Item Resource', () => {
+  let provider: DataProvider
+  let auth: AuthProvider
+
+  beforeAll(async () => {
+    const provider = await localForageDataProvider({
+      prefixLocalForageKey: TEST_STORAGE_KEY
+    })
+    for (const user of encryptedUsers) {
+      await provider.create<User>(R_USERS, { data: { ...user } })
+    }
+    auth = authProvider(provider)
+    await auth.login({ username: 'ian', password: process.env.PASSWORD })
+
+    // creating batch beacuse item_number is generated from batch
+    // in the aftercreate lifeCycle
+    await provider.create<Batch>(R_BATCHES, {
+      data: generateDummyBatchForTesting({ id: 1 })
+    })
+  })
+
+  beforeEach(async () => {
+    const withOutLifecycleProvider = await localForageDataProvider({
+      prefixLocalForageKey: TEST_STORAGE_KEY
+    })
+    provider = withLifecycleCallbacks(
+      withOutLifecycleProvider,
+      lifecycleCallbacks(
+        trackEvent(withOutLifecycleProvider),
+        withOutLifecycleProvider
+      )
+    )
+
```

Comment:
@taha - there are two very similar blocks in the next few lines.

I think we should declare an array of resource-names, then loop through the names clearing each.  We should use the same strategy in batch-test.

We will have tests for other resource types in the future, so we will just end up passing in the full list of resource names.

Reply:
You have the wrong Taha there 😅